doc/api/cli.md

```diff
@@ -871,6 +871,22 @@ added: v11.8.0
 
 Use the specified file as a security policy.
 
+### `--experimental-require-module`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active Developement
+
+Supports loading a synchronous ES module graph in `require()`. If the module
+graph is not synchronous (contains top-level await), it throws an error.
+
+By default, a `.js` file will be parsed as a CommonJS module first. If it
+contains ES module syntax, Node.js will try to parse and evaluate the module
+again as an ES module. If it turns out to be synchronous and can be evaluated
+successfully, the module namespace object will be returned by `require()`.
```

Comment:
We should do the detection behavior only for ambiguous files.

```suggestion
A `.js` file whose nearest parent `package.json` file contains no `"type"` field
will first be parsed as a CommonJS module. If it contains ES module syntax,
Node.js will try to parse and evaluate the module again as an ES module. If it
turns out to be synchronous and can be evaluated successfully, the module
namespace object will be returned by `require()`.
```

Reply:
As I said in the OP, the doc is basic, and not for review. I probably should've updated it to what I had in the OP

```
When the module being require()ed has .mjs extension or there are
other explicit indicators that it's an ES module, we load it as an
ES module. If the graph is synchronous, we return the module namespace
as the exports. If the graph contains top-level await, we throw an
error before evaluating the module. If an additional flag
--print-pending-tla is passed, we proceed to evaluation but do not
run the microtasks, only to find out where the TLAs are and print
their location to help users fix them.

If there are not explicit indicators whether a .js file is CJS or ESM,
we parse it as CJS first. If the parse error indicates that it contains
ESM syntax, we parse it again as ESM. If the second parsing succeeds,
we continue to treat it as ESM.
```

Reply:
Sure, that works, or even just TODO. I just wanted to ensure that our design intent is to run detection only for ambiguous files, like how `--experimental-detect-module` works.